### ISSUE TITLE
default configuration doesn't reduce size in Nextjs project.

### ISSUE THREAD

**Issue by pyprism** [FullHuman/purgecss#698]
I tried to use purgecss in a Nextjs project. I followed the documentation . But after setup I could not see any differences. Here is the project  package.json:

```
{
  "name": "****",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "size": "ANALYZE=true next build",
    "start": "next start -p 4321"
  },
  "dependencies": {
    "@fullhuman/postcss-purgecss": "^4.0.3",
    "@next/bundle-analyzer": "^11.0.0",
    "axios": "^0.21.1",
    "luxon": "^1.27.0",
    "next": "11.0.0",
    "next-pwa": "^5.2.21",
    "next-seo": "^4.25.1",
    "postcss-flexbugs-fixes": "^5.0.2",
    "postcss-preset-env": "^6.7.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-responsive-carousel": "^3.2.18"
  },
  "devDependencies": {
    "@babel/core": "^7.14.6",
    "@babel/preset-env": "^7.14.5",
    "bulma": "^0.9.2",
    "purgecss": "^4.0.3",
    "sass": "^1.35.1",
    "webpack": "^5.39.0"
  }
}
```

next.config.js :

```
const withPWA = require('next-pwa');
const path = require('path');

const withBundleAnalyzer = require('@next/bundle-analyzer')({
    enabled: process.env.ANALYZE === 'true',
})

let domain = '****';


let nextConfig = {
    target: "serverless",
    sassOptions: {
        includePaths: [path.join(__dirname, 'styles')],
    },
    env: {
        domain: domain,
    },
    images: {
        domains: ['***'],
    },
    pwa: {
        dest: 'public',
    }
}


if(process.env.DEBUG) {
    domain = 'http://localhost:8000';
    module.exports = withBundleAnalyzer({
        env: {
            domain: domain,
        },
        images: {
            domains: ['*'],
        },

    })
}

module.exports = withPWA(nextConfig);
```
postcss.config.js:

```
module.exports = {
    "plugins": [
        "postcss-flexbugs-fixes",
        [
            "postcss-preset-env",
            {
                "autoprefixer": {
                    "flexbox": "no-2009"
                },
                "stage": 3,
                "features": {
                    "custom-properties": false
                }
            }
        ],
        [
            '@fullhuman/postcss-purgecss',
            {
                content: [
                    './pages/**/*.{js,jsx,ts,tsx}',
                    './components/**/*.{js,jsx,ts,tsx}'
                ],
                defaultExtractor: content => content.match(/[\w-/:]+(?<!:)/g) || [],
                safelist: ["html", "body"]
            }
        ],
    ]
}
```

screenshot of before installing purgecss:
![image](https://user-images.githubusercontent.com/1469242/122514162-513d2700-d02d-11eb-82ff-ecee21d4e6b8.png)

after installing purgecss:

![image](https://user-images.githubusercontent.com/1469242/122514130-42567480-d02d-11eb-9c70-35aaa2a47502.png)

**Comment by antonioOrtiz** [comment]
+1

I'm experiencing the same problem:

package.json:


```
{
    "name": "hillfinder",
    "scripts": {
        "dev": "NODE_ENV=development nodemon server/index.js",
        "build": "next build",
        "start": "next start -p $PORT"
    },
    "dependencies": {
        "@fullhuman/postcss-purgecss": "^3.1.3",
        "@zeit/next-sass": "^1.0.1",
        "axios": "^0.19.2",
        "babel-plugin-transform-class-properties": "^6.24.1",
        "bcrypt": "^3.0.8",
        "body-parser": "^1.18.3",
        "cloudinary": "^1.22.0",
        "cloudinary-core": "^2.8.2",
        "cloudinary-react": "^1.5.0",
        "compression": "^1.7.4",
        "connect-mongo": "^2.0.3",
        "cookie-parser": "^1.4.4",
        "cors": "^2.8.5",
        "dotenv": "^8.0.0",
        "email-validator": "^2.0.4",
        "eslint": "^7.30.0",
        "eslint-config-next": "^11.0.1",
        "esri-leaflet-geocoder": "^2.3.3",
        "express": "^4.16.4",
        "express-session": "^1.15.6",
        "express-validator": "^6.5.0",
        "flatted": "^3.1.0",
        "font-awesome": "^4.7.0",
        "helmet": "^3.22.0",
        "http-status-codes": "^1.3.2",
        "imagemin-mozjpeg": "^9.0.0",
        "imagemin-optipng": "^8.0.0",
        "indicative": "^7.4.4",
        "leaflet": "^1.7.1",
        "leaflet-geosearch": "^3.0.6",
        "leaflet-routing-machine": "^3.2.12",
        "leaflet.locatecontrol": "^0.72.0",
        "lodash": "^4.17.20",
        "logrocket": "^1.0.14",
        "mailgun-js": "^0.22.0",
        "mongoose": "^5.7.5",
        "morgan": "^1.10.0",
        "next": "^10.0.0",
        "next-images": "^1.4.0",
        "next-optimized-images": "^2.6.1",
        "next-redux-wrapper": "^3.0.0",
        "next-server": "^9.0.5",
        "node-sass": "^4.14.1",
        "passport": "^0.4.1",
        "passport-local": "^1.0.0",
        "postcss-flexbugs-fixes": "^5.0.2",
        "postcss-preset-env": "^6.7.0",
        "react": "^16.14.0",
        "react-dom": "^16.14.0",
        "react-leaflet": "^2.7.0",
        "react-leaflet-control": "^2.1.2",
        "react-leaflet-search": "^2.0.1",
        "react-leaflet-universal": "^2.2.1",
        "react-redux": "^5.0.6",
        "react-router": "^5.1.2",
        "react-router-dom": "^5.1.2",
        "redux": "^4.0.4",
        "redux-devtools-extension": "^2.13.8",
        "redux-logger": "^3.0.6",
        "redux-persist": "^6.0.0",
        "redux-thunk": "^2.3.0",
        "semantic-ui-css": "^2.4.1",
        "semantic-ui-react": "^0.86.0"
    },
    "devDependencies": {
        "@zeit/next-css": "^1.0.1",
        "babel-eslint": "^10.0.1",
        "css-loader": "^1.0.1",
        "eslint-config-airbnb": "^17.1.0",
        "eslint-config-prettier": "^4.1.0",
        "eslint-config-react-app": "^3.0.8",
        "eslint-plugin-flowtype": "^2.50.3",
        "eslint-plugin-import": "^2.16.0",
        "eslint-plugin-jsx-a11y": "^6.2.1",
        "eslint-plugin-prettier": "^3.0.1",
        "eslint-plugin-react": "^7.12.4",
        "eslint-plugin-react-hooks": "^2.3.0",
        "file-loader": "^3.0.1",
        "next-purgecss": "^4.0.0",
        "nodemon": "^1.19.1",
        "prettier": "^1.16.4",
        "style-loader": "^0.23.1",
        "url-loader": "^1.1.2",
        "webpack": "^4.29.6"
    }
}
```
Not sure about included my `next.config.js` as it's just wiring my `.env` variables to PROD. And because `next 10.0.0` the version I'm using is doing so much under the covers! (css, sass, font-loading etc).

next.config.js:

```
module.exports = {
  env: {
    MONGODB_URI: process.env.MONGODB_URI,
    DEVELOPMENT_DB_DSN: process.env.DEVELOPMENT_DB_DSN,
    EMAIL_ADDRESS: process.env.EMAIL_ADDRESS,
    EMAIL_PASSWORD: process.env.EMAIL_PASSWORD,

    MAILGUN_DOMAIN: process.env.MAILGUN_DOMAIN,
    MAILGUN_API_KEY: process.env.MAILGUN_API_KEY,
    MAILGUN_PUBLIC_KEY: process.env.MAILGUN_PUBLIC_KEY,

    MAPBOX_USERNAME: process.env.MAPBOX_USERNAME,
    MAPBOX_STYLE_ID: process.env.MAPBOX_STYLE_ID,
    MAPBOX_ACCESS_TOKEN: process.env.MAPBOX_ACCESS_TOKEN,

    MAILGUN_SMTP_CREDENTIALS: process.env.MAILGUN_SMTP_CREDENTIALS,

    CLOUDINARY_API_KEY: process.env.CLOUDINARY_API_KEY,
    CLOUDINARY_API_SECRET: process.env.CLOUDINARY_API_SECRET,
    CLOUDINARY_CLOUD_NAME: process.env.CLOUDINARY_CLOUD_NAME,

    CLOUDINARY_URL: process.env.CLOUDINARY_URL
  }
};

```
My `postcss.config.js`:

```
module.exports = {
  plugins: [
    'postcss-flexbugs-fixes',
    [
      'postcss-preset-env',
      {
        autoprefixer: {
          flexbox: 'no-2009'
        },
        stage: 3,
        features: {
          'custom-properties': false
        }
      }
    ],
    [
      '@fullhuman/postcss-purgecss',
      {
        content: ['./pages/**/*.{js,jsx,ts,tsx}', './components/**/*.{js,jsx,ts,tsx}'],
        defaultExtractor: content => content.match(/[\w-/:]+(?<!:)/g) || [],
        safelist: ['html', 'body']
      }
    ]
  ]
};
```

And these are my logs—one with the `postcss.config.js` and without—can't imagine 1M being  the correct output?

![Screen Shot 2021-07-12 at 9 53 35 AM](https://user-images.githubusercontent.com/4282013/125299921-933d4e00-e2f7-11eb-9f46-28eb32166a20.jpg)

Any help would be appreciated!

**Comment by Ffloriel** [comment]
It seems to work with a new nextjs project.
This might be related to the nextjs cache: https://github.com/vercel/next.js/issues/31441
